dask_expr/_collection.py

```diff
@@ -5781,12 +5781,134 @@ def from_map(
     enforce_metadata=False,
     **kwargs,
 ):
-    """Create a dask-expr collection from a custom function map
+    """Create a DataFrame collection from a custom function map.
 
-    NOTE: The underlying ``Expr`` object produced by this API
-    will support column projection (via ``simplify``) if
-    the ``func`` argument has "columns" in its signature.
+    ``from_map`` is the preferred option when reading from data sources
+    that are not natively supported by Dask or if the data source
+    requires custom handling before handing things of to Dask DataFrames.
+    Examples are things like binary files or other unstructured data that
+    doesn't have an IO connector
 
+    ``from_map`` supports column projection by the optimizer. The optimizer
+    tries to push column selections into the from_map call if the function
+    supports a ``columns`` argument.
```

Comment:
Is it actually sufficient to have a columns argument? The paramter description below states that it has to implement the `DataFrameIOFunction` protocol, One of the two docs should be adjusted

Reply:
adjusted